framework/src/mesh/MooseMesh.C

```diff
@@ -3199,3 +3211,6 @@ MooseMesh::buildFaceInfo()
     }
   }
 }
+
+template ReplicatedMesh MooseMesh::buildTypedMesh(unsigned int);
+template DistributedMesh MooseMesh::buildTypedMesh(unsigned int);
```

Comment:
I'm not sure I'm a fan of the implementation file template instantiation technique.  It makes for unpleasant compiler errors when someone tries to use the template with a type that isn't supported/manually-instantiated.  I'd probably just prefer a header template impl that throws an error with specializations for supported types.  But this isn't a big deal - it is consistent with other places in the code.  Just my 2c.